common/xbps-src/shutils/update_check.sh

```diff
@@ -75,11 +76,19 @@ update_check() {
         rx=${pattern:-$rx}
         rx=${rx:-'(?<!-)\b\Q'"$pkgname"'\E[-_]?((src|source)[-_])?\K([^-/_\s]*?\d[^-/_\s]*?)(?=(?:[-_.](?:src|source|orig))?\.(?:[jt]ar|shar|t[bglx]z|tbz2|zip))\b'}
 
-        if [ -n "$XBPS_UPDATE_CHECK_VERBOSE" ]; then
-            echo "fetching $url" 1>&2
+        if printf "$fetchedurls" | grep -qFx "$url"; then
+            if [ -n "$XBPS_UPDATE_CHECK_VERBOSE" ]; then
+                echo "already fetched $url" 1>&2
+            fi
+            continue
+        else
+            if [ -n "$XBPS_UPDATE_CHECK_VERBOSE" ]; then
+                echo "fetching $url" 1>&2
+            fi
         fi
         curl -H 'Accept: text/html,application/xhtml+xml,application/xml,text/plain,application/rss+xml' -A "xbps-src-update-check/$XBPS_SRC_VERSION" --max-time 10 -Lsk "$url" |
             grep -Po -i "$rx"
+        fetchedurls="$fetchedurls\n$url"
```

Comment:
Not sure that works:
```
$ a="Hello"
$ a="$a\nWorld"
$ echo "$a"
Hello\nWorld
```
There's a bashism for that:
```
$ a="Hello"
$ a="$a"$'\n'"World"
$ echo "$a"
Hello
World
```
But not sure if we want that. Opinions?

Reply:
Happens to work because variable is used with `printf`, not `echo`, but it will break on `%` in url. 

What about literal newline and echo?
```        
        fetchedurls="$fetchedurls
$url"
echo "$fetchedurls" | …
```

Reply:
printf is actually a good idea:
```
$ a=$(print "%s\n%s" "$a" "$b")
```